server/sonar-server/src/main/java/org/sonar/server/db/migrations/v51/FeedIssueComponentUuids.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.db.migrations.v51;
+
+import org.sonar.core.persistence.Database;
+import org.sonar.server.db.migrations.BaseDataChange;
+import org.sonar.server.db.migrations.MassUpdate;
+import org.sonar.server.db.migrations.MassUpdate.Handler;
+import org.sonar.server.db.migrations.Select.Row;
+import org.sonar.server.db.migrations.SqlStatement;
+
+import java.sql.SQLException;
+
+public class FeedIssueComponentUuids extends BaseDataChange {
+
+  public FeedIssueComponentUuids(Database db) {
+    super(db);
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    MassUpdate update = context.prepareMassUpdate().rowPluralName("issues");
+    update.select(
+      "SELECT c.uuid, c.project_uuid, i.id " +
+        "FROM issues i " +
+        "INNER JOIN projects c ON i.component_id=c.id " +
+        "WHERE i.component_uuid is null");
```

Comment:
There are two strategies when optimizing reentrancy of migration:
- restrict input to non-updated rows in SQL request -> "where i.component_uuid is null"
- full scan table and post-filter in Java (remove "where i.component_uuid is null")

The first option can be faster if there's an index on the column, else execution time can be much slower than full scan. On the other hand, volume of response on the db wire is still smaller with first option.

All that to say: there's an index on ISSUES.COMPONENT_ID, so it sounds good !
But... this index is missing on the new column ISSUES.COMPONENT_UUID. Is it required regarding to existing SQL requests ?